### DISCUSSION TITLE
Is there a way to use relative paths for the Test Runner?

### DISCUSSION THREAD

**Discussion by pprotas** [Shopify/ruby-lsp#2965] [Q&A] [answered]
When I run a test in the VSCode extension it's executed using the absolute path. Our tests run in a Docker container due to our setup, so this results in the following error:
```
cannot load such file -- /Users/my-user/...../....._spec.rb
```

Is there a way to run the spec using a relative path so that it works with the Docker container?

**Comment by vinistock** [comment]
Did you already try to configure VS Code to connect to the container ([docs](https://github.com/Shopify/ruby-lsp/tree/main/vscode#developing-on-containers))?

**Comment by d-lebed** [comment]
I do not want to be annoying, but that's another problem with dockerized env, that can be addressed by proposed solution: https://github.com/Shopify/ruby-lsp/pull/2919

**Comment by pprotas** [comment, marked as answer]
Workaround with a wrapper script:

Create `rspec_wrapper` and then `chmod +x rspec_wrapper`:

```bash
#!/bin/bash

# Get the absolute path from the argument
ABS_PATH="$1"

# Get the current working directory
CURRENT_DIR="$PWD"

# Remove the current directory prefix from the absolute path
REL_PATH="${ABS_PATH#"$CURRENT_DIR"/}"

# Run the command with the relative path
bin/rspec "$REL_PATH"
```

Then in your `settings.json`:
```json
"rubyLsp.addonSettings": {
  "Ruby LSP Rails": {
    "rspecCommand": "./rspec_wrapper"
  }
},
```

Adjust the `rspec_wrapper` script to your liking. For example, to pass the path to some docker container:

```bash
docker exec -t my-container bin/rspec "$REL_PATH"
```

**Reply by Splines** [reply]
`rspecCommand` doesn't seem to be a valid configuration setting for the addon. See the [available settings in the docs](https://shopify.github.io/ruby-lsp/rails-add-on.html#settings). Maybe you meant [`Ruby LSP RSpec`](https://github.com/st0012/ruby-lsp-rspec) instead of `Ruby LSP Rails`?
